### PR TITLE
DDCNL-8997: Replace http-caching-client in pertax-frontend

### PR DESCRIPTION
**Pull request by askar-makkunnath**
- [x] UT Passed
- [x] IT Passed
- [x] AT Passed
- [x] Smoke Test Verified

### CODE REVIEW

File: app/connectors/AgentClientAuthorisationConnector.scala
Comment (pascal22p) [inline app/connectors/AgentClientAuthorisationConnector.scala:1]: It's a connector. it should not use UserAnwers. Revert the changes

File: app/repositories/SessionCacheRepository.scala
Comment (pascal22p) [inline app/repositories/SessionCacheRepository.scala:1]: Keep this repo for caching responses from connectors

File: app/models/models.scala
Comment (pascal22p) [inline app/models/models.scala:1]: Needs unit tests

File: app/models/AddressCacheIdentifiers.scala
Comment (pascal22p) [inline app/models/AddressCacheIdentifiers.scala:1]: Double check but I don't think this is part of any user journey?

**Review by pascal22p — CHANGES_REQUESTED** [review]

File: app/config/HmrcModule.scala [hunk 1]
```diff
@@ -19,13 +19,13 @@ package config
...
-import services.LocalSessionCache
-import uk.gov.hmrc.http.cache.client.SessionCache
+
+import java.time.{Clock, ZoneOffset}
 
 class HmrcModule extends Module {
   override def bindings(environment: Environment, configuration: Configuration): Seq[Binding[_]] = {
     val defaultBindings: Seq[Binding[_]] = Seq(
-      bind[SessionCache].to[LocalSessionCache],
+      bind[Clock].toInstance(Clock.systemDefaultZone.withZone(ZoneOffset.UTC)),
```
[the remark this answers is not in the document]
Comment (askar-makkunnath) [reply app/config/HmrcModule.scala]: Updated to use  Europe/London.

File: app/repositories/SessionCacheRepository.scala
Comment (askar-makkunnath) [reply app/repositories/SessionCacheRepository.scala:1]: Updated

File: app/models/models.scala
Comment (askar-makkunnath) [reply app/models/models.scala:1]: Added test/models/ModelsSpec.scala and test/models/UserAnswersSpec.scala

File: app/models/AddressCacheIdentifiers.scala
Comment (askar-makkunnath) [reply app/models/AddressCacheIdentifiers.scala:1]: This has been replaced by pages in  /routePages.

File: a11y/views/testSpec.scala [hunk 2]
```diff
@@ -203,32 +205,21 @@ class testSpec extends A11ySpec {
 
     server.stubFor(
       put(urlMatching(s"/keystore/pertax-frontend/.*"))
```
Comment (pascal22p) [inline a11y/views/testSpec.scala:207]: is keystore still used? I think only http-caching was using keystore.

File: app/controllers/controllershelpers/AddressJourneyCachingHelper.scala
Comment (pascal22p) [inline app/controllers/controllershelpers/AddressJourneyCachingHelper.scala:1]: This file needs some attention.
The old cache was used for both the user input and as a data cache for address-lookup. This does not feel right.
User inputs should be segregated from other data type.

I am not sure if this can be fixed in this ticket. Ideally address-lookup cache should be at the connector level

File: app/controllers/controllershelpers/HomePageCachingHelper.scala
Comment (pascal22p) [inline app/controllers/controllershelpers/HomePageCachingHelper.scala:1]: Can you check what this is used for? which banners?

**Review by pascal22p — CHANGES_REQUESTED** [review]

File: app/config/PertaxModule.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+package config
+
+import com.google.inject.AbstractModule
+
+import java.time.{Clock, ZoneId}
+
+class PertaxModule extends AbstractModule {
+  override def configure(): Unit =
+    bind(classOf[Clock]).toInstance(Clock.systemDefaultZone.withZone(ZoneId.of("Europe/London")))
+}
```
[the remark this answers is not in the document]
Comment (askar-makkunnath) [reply app/config/PertaxModule.scala]: Updated.

File: a11y/views/testSpec.scala (hunk 2 above)
Comment (askar-makkunnath) [reply a11y/views/testSpec.scala]: It was used only in test stubs. Removed it.

File: app/controllers/controllershelpers/HomePageCachingHelper.scala
Comment (askar-makkunnath) [reply app/controllers/controllershelpers/HomePageCachingHelper.scala:1]: Currently, this is used for storing UR banner dismissal status (through UserResearchDismissalController) and retrieving the dismissal status to be used for passing to HomeView. But retrieval is behind a toggle and corresponding value is set in ("feature.banner.home.enabled") config.

File: app/controllers/controllershelpers/AddressJourneyCachingHelper.scala
Comment (pascal22p) [reply app/controllers/controllershelpers/AddressJourneyCachingHelper.scala:1]: too big. Need new ticket

**Review by pascal22p — APPROVED** [review]
